tests/databases/constructed_molecule/get_all/test_get_all.py

```diff
@@ -0,0 +1,97 @@
+import stk
+
+from tests.utilities import is_equivalent_constructed_molecule
+
+
+def test_get_all(case_data):
+    """
+    Test iteration over all entries.
+
+    Parameters
+    ----------
+    case_data : :class:`.CaseData`
+        A test case. Holds the databases to test and the molecules to
+        place into the databases.
+
+    Returns
+    -------
+    None : :class:`NoneType`
+
+    """
+
+    _test_get_all(
+        inchi_database=case_data.inchi_database,
+        smiles_database=case_data.smiles_database,
+        inchi_key_database=case_data.inchi_key_database,
+        inchi_and_smiles_database=case_data.inchi_and_smiles_database,
+        molecules=case_data.molecules,
+    )
+
+
+def _test_get_all(
+    inchi_database,
+    smiles_database,
+    inchi_key_database,
+    inchi_and_smiles_database,
+    molecules,
+):
+    """
+    Test iteration over all entries.
+
+    Parameters
+    ----------
+    inchi_database : :class:`.MoleculeDatabase`
+        A database to test.
+
+    smiles_database : :class:`.MoleculeDatabase`
+        A database to test.
+
+    inchi_key_database : :class:`.MoleculeDatabase`
+        A database to test.
+
+    inchi_and_smiles_database : :class:`.MoleculeDatabase`
+        The database to test collection from.
+
+    molecules : :class:`iterable` of :class:`.Molecule`
+        The molecules to put and get from the databases.
+
+    Returns
+    -------
+    None : :class:`NoneType`
+
+    """
```

Comment:
This test needs to be reworked. There are 2 issues here

* Not every database will use inchi and smiles, or molecule keys at all. Having variables refer to them is leaking something specific to one database -> the test is not generic enough.
* All the putting of stuff into the a database is not the actual test, it is test setup. it should therefore be part of a fixture, not part of the test itself.

Reply:
I think the test should be, something along the lines of 


```python

def test_get_all(case_data: CaseData) -> None:
    _test_get_all(
        database=case_data.database,
        expected_molecules=case_data.expected_molecules
    )

def _test_get_all(...) -> None:
        smiles = stk.Smiles()
        for i, retrieved in enumerate(database.get_all()):
            expected = expected_molecules[smiles.get_key(retrieved)]
            is_equivalent_molecule(...)
        
        # the rest
```

THe only pieces of data the test-case should contain is the database we're testing and a map mapping the smiles to the expected molecule